nncf/experimental/tensor/torch_functions.py

```diff
@@ -0,0 +1,148 @@
+    TensorDataType.uint8: torch.uint8,
+}
+
+DTYPE_MAP_REV = {v: k for k, v in DTYPE_MAP.items()}
+
+
+@functions.device.register(torch.Tensor)
+def _(a: torch.Tensor) -> TensorDeviceType:
+    DEVICE_MAP = {
+        "cpu": TensorDeviceType.CPU,
+        "cuda": TensorDeviceType.GPU,
+    }
+    return DEVICE_MAP[a.device.type]
+
+
+@functions.squeeze.register(torch.Tensor)
+def _(a: torch.Tensor, axis: Optional[Union[int, Tuple[int]]] = None) -> torch.Tensor:
+    if axis is None:
+        return a.squeeze()
+    return a.squeeze(axis)
+
+
+@functions.flatten.register(torch.Tensor)
+def _(a: torch.Tensor) -> torch.Tensor:
+    return a.flatten()
+
+
+@functions.max.register(torch.Tensor)
+def _(a: torch.Tensor, axis: Optional[Union[int, Tuple[int]]] = None) -> torch.Tensor:
+    if axis is None:
+        return torch.max(a)
+    return torch.max(a, dim=axis).values
+
+
+@functions.min.register(torch.Tensor)
+def _(a: torch.Tensor, axis: Optional[Union[int, Tuple[int]]] = None) -> torch.Tensor:
+    if axis is None:
+        return torch.min(a)
+    return torch.min(a, dim=axis).values
+
+
+@functions.abs.register(torch.Tensor)
+def _(a: torch.Tensor) -> torch.Tensor:
+    return torch.absolute(a)
+
+
+@functions.astype.register(torch.Tensor)
+def _(a: torch.Tensor, dtype: TensorDataType) -> torch.Tensor:
+    return a.type(DTYPE_MAP[dtype])
+
+
+@functions.dtype.register(torch.Tensor)
+def _(a: torch.Tensor) -> TensorDataType:
+    return DTYPE_MAP_REV[a.dtype]
+
+
+@functions.reshape.register(torch.Tensor)
+def _(a: torch.Tensor, shape: List[int]) -> torch.Tensor:
+    return a.reshape(shape)
+
+
+@functions.all.register(torch.Tensor)
+def _(a: torch.Tensor, axis: Optional[Union[int, Tuple[int]]] = None) -> Union[torch.Tensor, bool]:
+    if axis is None:
+        return torch.all(a)
+    return torch.all(a, dim=axis)
+
+
+@functions.allclose.register(torch.Tensor)
+def _(a: torch.Tensor, b: torch.Tensor, rtol: float = 1e-05, atol: float = 1e-08, equal_nan: bool = False) -> bool:
+    return torch.allclose(a, b, rtol=rtol, atol=atol, equal_nan=equal_nan)
+
+
+@functions.any.register(torch.Tensor)
+def _(a: torch.Tensor, axis: Optional[Union[int, Tuple[int]]] = None) -> Union[torch.Tensor, bool]:
+    if axis is None:
+        return torch.any(a)
+    return torch.any(a, dim=axis)
+
+
+@functions.count_nonzero.register(torch.Tensor)
+def _(a: torch.Tensor, axis: Optional[Union[int, Tuple[int]]] = None) -> torch.Tensor:
+    return torch.count_nonzero(a, dim=axis)
+
+
+@functions.isempty.register(torch.Tensor)
+def _(a: torch.Tensor) -> bool:
+    return a.numel() == 0
+
+
+@functions.isclose.register(torch.Tensor)
+def _(a: torch.Tensor, b: torch.Tensor, rtol: float = 1e-05, atol: float = 1e-08, equal_nan: bool = False):
+    return torch.isclose(a, b, atol=atol, rtol=rtol, equal_nan=equal_nan)
+
+
+@functions.maximum.register(torch.Tensor)
+def _(x1: torch.Tensor, x2: torch.Tensor) -> torch.Tensor:
+    if not isinstance(x2, torch.Tensor):
```

Comment:
Could you please explain why this check is required?

Reply:
Torch functions minimum and maximum required tensor as second input on same device
```python
functions.minimum(Tensor(torch.tensor([1,2]).cuda()), 0)    # nncf.Tensor(tensor([0, 0]))
```